Vineyard/Vineyard/CreateGoalView.swift

```diff
@@ -9,305 +9,111 @@ import SwiftUI
             }
+            .ignoresSafeArea(.keyboard)
         }
-        .onAppear {
-            if editMode {
-                populateFormFields()
-            }
-        }
-        .padding()
-        .toolbar(content: {
-            ToolbarItem(placement: .principal) {
-                
-            }
-        })
     }
-    
+
     func populateFormFields() {
-        goalName = goals[indexOfGoal].title // the name of the goal
-        description = goals[indexOfGoal].description // the name of the goal
-        quantity = String(goals[indexOfGoal].quantity ?? 0)
-        
-        selectedDifficulty = goals[indexOfGoal].diffLevel.difficultyLevel
-        selectedFrequency = goals[indexOfGoal].frequency.frequencyType
-        selectedFrequencyText = goals[indexOfGoal].frequency.frequencyType.rawValue
-        freqQuantity = 0
+        resolutionResult.resolutionTitle = goals[indexOfGoal].title // the name of the goal
+        resolutionResult.extractedQuantity = goals[indexOfGoal].quantity
+        hasQuantity = goals[indexOfGoal].quantity != nil
+
+        resolutionResult.difficulty = goals[indexOfGoal].diffLevel.difficultyLevel
+        resolutionResult.frequencyType = goals[indexOfGoal].frequency.frequencyType
+        resolutionResult.frequencyQuantity = goals[indexOfGoal].frequency.count
     }
     
     func submitGoalCreationForm() {
         do {
-            try viewModel.validateGoalCreationForm(action: goalName, quantity: quantity, isQuantityTask: isQuantityTask, isInserted: isInserted)
-
-            if let index = indexInserted {
-                words.insert("qtt_position", at: index)
+            try viewModel
+                .validateGoalCreationForm(
+                    action: resolutionResult.resolutionTitle,
+                    quantity: resolutionResult.extractedQuantity,
+                    isQuantityTask: /*resolutionResult.hasQuantity*/ resolutionResult.extractedQuantity != nil
+                )
 
-            }
-            goalName = words.joined(separator: " ")
             if editMode {
-                goals[indexOfGoal] = Resolution(id: UUID().uuidString, title: goalName, description: description, quantity: Int(quantity), frequency: Frequency(frequencyType: selectedFrequency, count: freqQuantity), diffLevel: Difficulty(difficultyLevel: selectedDifficulty, score: 10))
+                goals[indexOfGoal] = Resolution(
+                    id: UUID().uuidString,
+                    title: resolutionResult.resolutionTitle,
+                    description: /*description*/ "",
+                    quantity: resolutionResult.extractedQuantity,
+                    frequency: Frequency(
+                        frequencyType: resolutionResult.frequencyType,
+                        count: resolutionResult.frequencyQuantity
+                    ),
+                    diffLevel: Difficulty(
+                        difficultyLevel: resolutionResult.difficulty,
+                        score: 10
```

Comment:
can you make this dependent on the diff level instead of a fixed score. Easy = 100, Med = 200, Diff = 300

Reply:
doing in another PR

Reply:
is this 100/200/300 fixed or can that be changed?

Reply:
issue #102

Reply:
I think we leave it fixed for now, it makes it the easisest. we could possibly make it so that in the group creation process you set the easy medium and difficult values. I think for demo day we just keep it fixed